app/internal/app/certificate_provider_store.go

```diff
@@ -0,0 +1,66 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+)
+
+type certificateProviderStore struct {
+	dataStore DataStore
+	now       func() time.Time
+}
+
+func (s *certificateProviderStore) Create(ctx context.Context) (*actor.CertificateProvider, error) {
+	data, err := page.SessionDataFromContext(ctx)
+	if err != nil {
+		return &actor.CertificateProvider{}, err
+	}
+
+	cp := &actor.CertificateProvider{LpaID: data.LpaID}
+	err = s.Put(ctx, cp)
+
+	return cp, err
+}
+
+func (s *certificateProviderStore) Get(ctx context.Context) (*actor.CertificateProvider, error) {
+	data, err := page.SessionDataFromContext(ctx)
+	if err != nil {
+		return &actor.CertificateProvider{}, err
+	}
+
+	if data.LpaID == "" {
+		return &actor.CertificateProvider{}, errors.New("certificateProviderStore.Get requires LpaID to retrieve")
+	}
+
+	var certificateProvider actor.CertificateProvider
+
+	pk := "LPA#" + data.LpaID
+
+	if err := s.dataStore.GetOneByPartialSk(ctx, pk, "#CERTIFICATE_PROVIDER#", &certificateProvider); err != nil {
```

Comment:
I was trying to keep this flexible so we can retrieve the CP when the donor is logged in as we won't have the session ID in that instance. This may not be an issue if we implement your suggestion on a store that brings everything back though.